services/batch_manager.js

```diff
@@ -18,27 +18,34 @@ const sequelize = new Sequelize(dbConfig.database, dbConfig.username, dbConfig.p
         }).catch(() => {
             throw new Api500Error(`Error in batch search`)
         })
         return results;
 
-    } else {
-        const results = await sequelize.query("SELECT batches.id , batches.price ,batches.start_time, batches.end_time,batches.days , academy.id AS academy_id, academy.name AS academy_name, arena.name AS arena_name,arena.locality,arena.city,arena.state,coaches.id AS coach_id, coaches.name AS coach_name , coaches.experience, sports.id AS sport_id, sports.name AS sport_name,batches.thumbnail_img AS batch_thumbnail,sports.type, (6371 *acos(cos(radians(" + req.query.lat + ")) * cos(radians(lat)) * cos(radians(lng) - radians(" + req.query.lng + ")) + sin(radians(" + req.query.lat + ")) * sin(radians(lat)))) AS distance FROM ((((Batches batches INNER JOIN Arenas arena on batches.arena_id = arena.id) INNER JOIN Coaches coaches ON batches.coach_id = coaches.id) INNER JOIN Sports sports ON batches.sports_id = sports.id) LEFT JOIN Academies academy on batches.academy_id = academy.id) HAVING distance < 100000   ORDER BY distance LIMIT 0, 20;", { type: Sequelize.QueryTypes.SELECT }).then((result) => {
+    } else {const results = await sequelize.query("SELECT batches.id , batches.price ,batches.start_time, batches.end_time,batches.days , academy.id AS academy_id, academy.name AS academy_name, arena.name AS arena_name,arena.locality,arena.city,arena.state,arena.lat,arena.lng,coaches.id AS coach_id, coaches.name AS coach_name , coaches.experience, sports.id AS sport_id, sports.name AS sport_name,batches.thumbnail_img AS batch_thumbnail,sports.type FROM ((((Batches batches INNER JOIN Arenas arena on batches.arena_id = arena.id) INNER JOIN Coaches coaches ON batches.coach_id = coaches.id) INNER JOIN Sports sports ON batches.sports_id = sports.id) LEFT JOIN Academies academy on batches.academy_id = academy.id);", { type: Sequelize.QueryTypes.SELECT }).then((result) => {
             return result;
         }).catch(() => {
             throw new Api500Error(`Error in batch search`)
         })
         return results;
     }
 }
-exports.process_batch_search_input_req = async (input_response) => {
+exports.process_batch_search_input_req = async (req,resp,input_response) => {
     var processed_response = [], overall_ratings = 0, ratings;
     if (input_response == null) {
         return input_response
     }
+    for (each_input_response of input_response) {
+       var lng1 = req.query.lng;
```

Comment:
variables defined here are only used once hence directly pass the values in the range method instead of storing it in  a variable ad passing the vairable